lib/associations/mixin.js

```diff
@@ -366,7 +366,7 @@ Mixin.getAssociation = function(target, alias) {
     if (this.associations.hasOwnProperty(associationName)) {
       var association = this.associations[associationName];
 
-      if (association.target === target && (alias === undefined ? !association.isAliased : association.as === alias)) {
+      if (association.target.name === target.name && (alias === undefined ? !association.isAliased : association.as === alias)) {
```

Comment:
Since target may be a scoped model, created with `Object.create()` we can't use strict equality